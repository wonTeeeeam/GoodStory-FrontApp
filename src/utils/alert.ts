import {Alert} from 'react-native';

export const alert = ({title = '', body = '', isConfirm = false}) => {
  if (isConfirm) {
    return new Promise(resolve => {
      Alert.alert(title, body, [
        {
          text: '확인',
          onPress: () => {
            resolve(true);
          },
        },
        {
          text: '취소',
          onPress: () => {
            resolve(false);
          },
        },
      ]);
    });
  }
  return new Promise(resolve => {
    Alert.alert(title, body, [
      {
        text: '확인',
        onPress: () => {
          resolve(true);
        },
      },
    ]);
  });
};
