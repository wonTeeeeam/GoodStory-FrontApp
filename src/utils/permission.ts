import {PERMISSIONS, check} from 'react-native-permissions';

export const checkNotificationPermission = async () => {
  const result = await check(PERMISSIONS.ANDROID.POST_NOTIFICATIONS);
  return result;
};
