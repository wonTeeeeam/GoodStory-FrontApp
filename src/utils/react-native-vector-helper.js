import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import FontAwesome5Pro from 'react-native-vector-icons/FontAwesome5Pro';
import FontAwesome6 from 'react-native-vector-icons/FontAwesome6';
import FontAwesome6Pro from 'react-native-vector-icons/FontAwesome6Pro';
import Fontisto from 'react-native-vector-icons/Fontisto';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Zocial from 'react-native-vector-icons/Zocial';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

export {
  AntDesign,
  Entypo,
  Feather,
  FontAwesome,
  FontAwesome5,
  FontAwesome5Pro,
  FontAwesome6,
  FontAwesome6Pro,
  Fontisto,
  Foundation,
  Ionicons,
  MaterialCommunityIcons,
  MaterialIcons,
  Octicons,
  Zocial,
  SimpleLineIcons,
};
