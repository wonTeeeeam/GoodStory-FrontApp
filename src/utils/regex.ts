const validateEmail = (value: string) => {
  const re =
    // (특수 기호들 제외 문자 + .다음에 특수기호 제외 문자) or 따옴표 안에 문자들 @ ip주소(123.123.123.123) or 영문자, 하이픈, 숫자 . 영문자 2개 이상
    // /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    // 영문자, 숫자로 시작한다, 영문자, 숫자, 하이픈, 언더바만 가능하다. @ 뒤에는 ip주소(123.123.123.123) or 영문자, 하이픈, 숫자 . 영문자 2개 이상
    /^[a-zA-Z0-9_-]{3,}[a-zA-Z0-9_-]*@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(value).toLowerCase());
};

const validateUserName = (input: string) => {
  try {
    const re = /^[가-힣|a-z|A-Z]{2,}$/;
    if (!re.test(input)) {
      return false;
    }
    return true;
  } catch (err) {
    return false;
    // return { error: err.message };
  }
};

const validatePWD = (password: string) => {
  // Minimum eight characters, at least one letter and one number:
  // const re = /^(?=.*[a-zA-Z])(?=.*\d).{8,}$/;

  // Minimum eight characters, at least one letter, one number and one special character:
  // const re = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;

  // 최소 영문자 1글자, 최소 특수문자(문자, 숫자, 공백이 아닌 것들) 또는 숫자 1글자, 공백 불가
  const re =
    /^(?=.*[A-Za-z])(?=.*([^\w\d\s]|[0-9]))(?!.*\s)([A-Za-z0-9]|[^\w\d\s]).{7,}$/;

  // 최소 영문자 1글자 + 숫자 + 특수문자 3가지 조합만 가능,  특수기호는 32가지(~․!@#$%^&*()_-+=[]{}|\;:‘“<>,.?/)만 허용, 공백불가, 영문자, 숫자, 특수문자 제외한 모든 것 불가, 최소 9글자
  // const re =
  //   /^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[~․!@#$%^&*()_\-+=[\]{}|\\;:‘“<>,.?/])(?!.*\s)(?!.*[^a-zA-Z0-9~․!@#$%^&*()_\-+=[\]{}|\\;:‘“<>,.?/]).{9,}$/;

  return re.test(password);

  // // Minimum eight characters, at least one uppercase letter, one lowercase letter and one number:
  // const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;

  // // Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character:
  // const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;

  // // Minimum eight and maximum 10 characters, at least one uppercase letter, one lowercase letter, one number and one special character:
  // const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,10}$/;
};

export {validateEmail, validatePWD, validateUserName};
