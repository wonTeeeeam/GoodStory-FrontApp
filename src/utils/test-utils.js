import React from 'react';
import {render} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import userReducer from 'slice/userSlice';
import navigationReducer from 'slice/navigationSlice';
import {setupStore} from 'store/store';

// As a basic setup, import your same slice reducers

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {},
) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
