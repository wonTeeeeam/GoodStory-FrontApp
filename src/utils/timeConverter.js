export function convertTimeToKorean(rawTime) {
  const inputTime = new Date(rawTime);
  const presentTime = Date.now();

  const gap = presentTime - inputTime;

  //   1분 미만
  if (gap <= 59000) {
    return '방금';
    // 60분 미만
  } else if (gap < 3600000) {
    return `${parseInt(gap / 60000, 10)}분`;
    // 24시간 미만
  } else if (gap < 86400000) {
    return `${parseInt(gap / 3600000, 10)}시간`;
  } else {
    return '1일 이상';
  }
}

export function convertTimeToStandardFormat(createdDate) {
  const inputDate = new Date(createdDate);
  const month = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];

  return (
    inputDate.getDate() +
    ' ' +
    month[inputDate.getMonth()] +
    ' ' +
    inputDate.getFullYear() +
    ' / ' +
    inputDate.getHours() +
    ':' +
    inputDate.getMinutes()
  );
}
