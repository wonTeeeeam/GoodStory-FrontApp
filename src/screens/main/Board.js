import React, {useEffect, useState} from 'react';
import {useNavigation} from '@react-navigation/native';
import {View, Text, StyleSheet} from 'react-native';
// import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

import FilterBar from '../../components/molecules/FilterBar';
import PostList from '../../components/organisms/PostList';

import {BackgroundColor} from '../../styles/BackgroundColor';
import {ss} from '../../utils/scailing';

export default function Board({route}) {
  const navigation = useNavigation();
  const [filterValue, setFilterValue] = useState('최신순');

  const changeFilterValue = newFilterValue => {
    setFilterValue(newFilterValue);
  };

  // const BottomTab = createBottomTabNavigator();

  return (
    <View style={styles.allContainer}>
      <View style={styles.filterBar}>
        <FilterBar
          filterValue={filterValue}
          changeFilterValue={changeFilterValue}
        />
      </View>
      <View style={styles.postList}>
        <PostList
          filterValue={filterValue}
          navigation={navigation}
          topic={route.params.boardTopic}
        />
      </View>
      {/* <BottomTab.Navigator>
        <BottomTab.Screen name="Home" component={Board} />
        {/* <BottomTab.Screen name="Settings" component={SettingsScreen} /> */}
      {/* </BottomTab.Navigator>  */}
    </View>
  );
}

const styles = StyleSheet.create({
  allContainer: {
    flex: 1,
  },
  filterBar: {
    flex: 0.04,
    backgroundColor: BackgroundColor.lightGray,
    justifyContent: 'center',
  },
  postList: {
    flex: 0.96,
    backgroundColor: BackgroundColor.snow,
  },
});
