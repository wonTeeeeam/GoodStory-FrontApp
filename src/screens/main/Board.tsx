import React, {useEffect, useState} from 'react';
import {View, Text, StyleSheet, BackHandler} from 'react-native';

import FilterBar from 'components/FilterBar';
import PostList from 'components/PostList';
import {gray, white} from 'styles';
import {showToast} from 'utils/toast';

export type Props = {
  route: {key: string; name: string; params: {boardTopic: string}};
};

const Board: React.FC<Props> = ({route}) => {
  const [filterValue, setFilterValue] = useState('최신순');
  const [backPressCount, setBackPressCount] = useState(0);

  const changeFilterValue = (newFilterValue: string) => {
    setFilterValue(newFilterValue);
  };

  useEffect(() => {
    const backAction = () => {
      setTimeout(() => {
        setBackPressCount(0);
      }, 2000);
      setBackPressCount(backPressCount + 1);
      if (backPressCount === 0) {
        showToast('한번 더 누르시면 앱이 종료됩니다');
      }
      if (backPressCount > 0) {
        BackHandler.exitApp();
      }
      return true;
    };

    const backHandler = BackHandler.addEventListener(
      'hardwareBackPress',
      backAction,
    );

    return () => {
      backHandler.remove();
    };
  }, [backPressCount]);

  return (
    <View style={styles.mainContainer}>
      <View style={styles.filterBarContainer}>
        <FilterBar
          filterValue={filterValue}
          changeFilterValue={changeFilterValue}
        />
      </View>
      <View style={styles.postListContainer}>
        <PostList filterValue={filterValue} topic={route.params.boardTopic} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
  },
  filterBarContainer: {
    flex: 0.04,
    backgroundColor: gray.lightGray,
    justifyContent: 'center',
  },
  postListContainer: {
    flex: 0.96,
    backgroundColor: white.snow,
  },
});

export default Board;
