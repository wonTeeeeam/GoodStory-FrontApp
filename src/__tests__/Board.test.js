import React from 'react';
import Board from 'screens/main/Board';
import {fireEvent, waitFor, within} from '@testing-library/react-native';
import {renderWithProviders} from 'utils/test-utils';

import {convertTimeToKorean} from 'utils/timeConverter';

const data = {
  data: [
    {
      BoardId: '7a23c7e8-3fd6-4c60-b5db-a426d410bb11',
      Category: 'Free',
      Created_date: '2023-07-02T05:46:27.466Z',
      Updated_date: '2023-07-08T08:27:26.087Z',
      Title: '1번게시글제목',
      Content: '1번게시글내용',
      Like: 0,
      Views: 38,
      ReplyCount: 6,
      user: {
        UserId: 'f3128064-28a4-47c2-bda4-9e1b5987300f',
        Nickname: '홍길동',
        Role: 'Admin',
        CompanyCode: '321321',
        CompanyName: '좆소',
        Created_date: '2023-03-31T12:24:45.043Z',
        Updated_date: '2023-07-11T12:03:44.205Z',
        Deleted_date: null,
        ProfilePhoto:
          'https://good-story.s3.ap-northeast-2.amazonaws.com/Profile/qksdnjswo@gmail.com/profile.png',
      },
      BoardPhotos: [],
    },
    {
      BoardId: '1fe34e61-ae32-4d59-b6ab-f69029e26a6b',
      Category: 'Free',
      Created_date: '2023-06-11T15:17:24.875Z',
      Updated_date: '2023-07-08T08:29:06.706Z',
      Title: '2번게시글제목',
      Content: '2번게시글내용',
      Like: 1,
      Views: 173,
      ReplyCount: 5,
      user: {
        UserId: 'f3128064-28a4-47c2-bda4-9e1b5987300f',
        Nickname: '홍길동',
        Role: 'Admin',
        CompanyCode: '321321',
        CompanyName: '좆소',
        Created_date: '2023-03-31T12:24:45.043Z',
        Updated_date: '2023-07-11T12:03:44.205Z',
        Deleted_date: null,
        ProfilePhoto:
          'https://good-story.s3.ap-northeast-2.amazonaws.com/Profile/qksdnjswo@gmail.com/profile.png',
      },
      BoardPhotos: [
        {
          BoardPhotoID: '51322338-3268-42c9-a131-99d8a70e90a5',
          URL: 'https://good-story.s3.ap-northeast-2.amazonaws.com/Board/1fe34e61-ae32-4d59-b6ab-f69029e26a6b/boardPhoto0.png',
          Created_date: '2023-04-29T11:19:13.053Z',
        },
      ],
    },
    {
      BoardId: '1882a0b6-329c-4a5e-a53c-63f8d84dfe8f',
      Category: 'Free',
      Created_date: '2023-05-12T03:19:02.596Z',
      Updated_date: '2023-07-02T05:33:57.655Z',
      Title: '3번게시글제목',
      Content: '3번게시글내용',
      Like: 0,
      Views: 23,
      ReplyCount: 0,
      user: {
        UserId: 'f3128064-28a4-47c2-bda4-9e1b5987300f',
        Nickname: '홍길동',
        Role: 'Admin',
        CompanyCode: '321321',
        CompanyName: '좆소',
        Created_date: '2023-03-31T12:24:45.043Z',
        Updated_date: '2023-07-11T12:03:44.205Z',
        Deleted_date: null,
        ProfilePhoto:
          'https://good-story.s3.ap-northeast-2.amazonaws.com/Profile/qksdnjswo@gmail.com/profile.png',
      },
      BoardPhotos: [],
    },
    {
      BoardId: '5edf689e-fb45-4bb0-876c-aefa95be7e6b',
      Category: 'Free',
      Created_date: '2023-05-12T03:09:47.283Z',
      Updated_date: '2023-07-08T08:23:18.708Z',
      Title: '4번게시글제목',
      Content: '4번게시글내용',
      Like: 0,
      Views: 6,
      ReplyCount: 0,
      user: {
        UserId: 'f3128064-28a4-47c2-bda4-9e1b5987300f',
        Nickname: '홍길동',
        Role: 'Admin',
        CompanyCode: '321321',
        CompanyName: '좆소',
        Created_date: '2023-03-31T12:24:45.043Z',
        Updated_date: '2023-07-11T12:03:44.205Z',
        Deleted_date: null,
        ProfilePhoto:
          'https://good-story.s3.ap-northeast-2.amazonaws.com/Profile/qksdnjswo@gmail.com/profile.png',
      },
      BoardPhotos: [],
    },
    {
      BoardId: '7926db7a-8bdf-409e-bfdf-cc90bbad5694',
      Category: 'Free',
      Created_date: '2023-04-29T12:41:10.536Z',
      Updated_date: '2023-06-12T15:11:32.714Z',
      Title: '5번게시글제목',
      Content: '5번게시글내용',
      Like: 0,
      Views: 7,
      ReplyCount: 0,
      user: {
        UserId: 'f3128064-28a4-47c2-bda4-9e1b5987300f',
        Nickname: '홍길동',
        Role: 'Admin',
        CompanyCode: '321321',
        CompanyName: '좆소',
        Created_date: '2023-03-31T12:24:45.043Z',
        Updated_date: '2023-07-11T12:03:44.205Z',
        Deleted_date: null,
        ProfilePhoto:
          'https://good-story.s3.ap-northeast-2.amazonaws.com/Profile/qksdnjswo@gmail.com/profile.png',
      },
      BoardPhotos: [
        {
          BoardPhotoID: '823f93b0-e694-4529-8e20-9e1ad236806d',
          URL: 'https://good-story.s3.ap-northeast-2.amazonaws.com/Board/7926db7a-8bdf-409e-bfdf-cc90bbad5694/boardPhoto0.png',
          Created_date: '2023-04-29T12:41:10.523Z',
        },
      ],
    },
  ],
};

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve(data)),
  post: jest.fn(() => Promise.resolve(true)),
  delete: jest.fn(() => Promise.resolve(true)),
}));

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity',
  () => 'TouchableOpacity',
);

const route = {
  key: 'Board-MAkMB4XImCVfcaiQH-caF',
  name: 'Board',
  params: {
    boardTopic: 'Free',
  },
};

describe('Board 테스트', () => {
  it('스냅샷 테스트', async () => {
    const tree = renderWithProviders(<Board route={route} />);
    await waitFor(() => expect(tree).toMatchSnapshot());
  });

  it('필터링 바 정상 렌더링 && 정상작동 테스트', async () => {
    const {queryByText} = renderWithProviders(<Board route={route} />);
    // 최초 filterValue는 '최신순'
    await waitFor(() => expect(queryByText('최신순')).toBeTruthy());

    // filterValue를 '추천순'으로 변경합니다.
    fireEvent.press(queryByText('최신순'));
    fireEvent.press(queryByText('추천순'));

    // filterValue가 '추천순'으로 바뀌었는지 체크합니다.
    await waitFor(() => expect(queryByText('추천순')).toBeTruthy());

    // filterValue를 다시 '최신순'으로 변경합니다.
    fireEvent.press(queryByText('추천순'));
    fireEvent.press(queryByText('최신순'));
    await waitFor(() => expect(queryByText('최신순')).toBeTruthy());
  });

  it('게시글 정상 렌더링 테스트', async () => {
    const {queryAllByTestId, debug} = renderWithProviders(
      <Board route={route} />,
    );

    await waitFor(() => {
      const renderedItems = queryAllByTestId('flatListItems');
      // 총 게시글 길이가 맞는지 체크
      expect(renderedItems).toHaveLength(data.data.length);
      renderedItems.forEach((item, index) => {
        const {queryByText, queryByTestId} = within(item);
        // 게시글 카테고리, 시간 잘 나오는가?
        expect(queryByText(data.data[index].Category)).toBeTruthy();
        expect(
          queryByText(convertTimeToKorean(data.data[index].Created_date)),
        ).toBeTruthy();

        // 게시글 작성자 회사, 닉네임 잘 나오는가?
        expect(queryByText(data.data[index].user.CompanyName)).toBeTruthy();
        expect(queryByText(data.data[index].user.Nickname)).toBeTruthy();

        // 게시글 제목, 내용, 사진 잘 나오는가?
        expect(queryByText(data.data[index].Title)).toBeTruthy();
        expect(queryByText(data.data[index].Content)).toBeTruthy();
        if (data.data[index].BoardPhotos[0]) {
          expect(queryByTestId('boardImage').props.source.uri).toBe(
            data.data[index].BoardPhotos[0].URL,
          );
        }

        // 좋아요, 조회수, 댓글 개수 잘 나오는가?
        expect(
          queryByText(
            data.data[index].Like ? data.data[index].Like.toString() : '좋아요',
          ),
        ).toBeTruthy();

        expect(
          queryByText(
            data.data[index].Views
              ? data.data[index].Views.toString()
              : '조회수',
          ),
        ).toBeTruthy();

        expect(
          queryByText(
            data.data[index].ReplyCount
              ? data.data[index].ReplyCount.toString()
              : '댓글',
          ),
        ).toBeTruthy();
      });
    });
  });

  it('좋아요 변경 테스트', async () => {
    const {queryAllByTestId, rerender, debug} = renderWithProviders(
      <Board route={route} />,
      {
        preloadedState: {
          user: {
            userId: 'asdasd',
            account: '',
            nickName: '',
            profileImage: '',
            createdDate: '',
            companyCode: '',
            companyName: '',
            accessToken: '',
            likeBoards: [],
            likeReReplies: [],
            likeReplies: [],
          },
        },
      },
    );
    await waitFor(() => {
      const renderedItems = queryAllByTestId('flatListItems');
      expect(renderedItems).toHaveLength(data.data.length);
      renderedItems.forEach((item, index) => {
        const {queryByText} = within(item);

        // 좋아요 플러스 작동
        fireEvent.press(
          queryByText(
            data.data[index].Like ? data.data[index].Like.toString() : '좋아요',
          ),
        );
      });
    });
    // 좋아요 state의 변화로 rerender
    rerender(<Board route={route} />);

    await waitFor(() => {
      const renderedItems = queryAllByTestId('flatListItems');
      expect(renderedItems).toHaveLength(data.data.length);
      renderedItems.forEach((item, index) => {
        const {queryByText} = within(item);
        // 좋아요 1개씩 올라갔는지 테스트
        const likeButton = queryByText((data.data[index].Like + 1).toString());
        expect(likeButton).toBeTruthy();

        // 좋아요 다시 취소하기.
        fireEvent.press(likeButton);
      });
    });

    // 좋아요 state의 변화로 rerender
    rerender(<Board route={route} />);

    await waitFor(() => {
      const renderedItems = queryAllByTestId('flatListItems');
      expect(renderedItems).toHaveLength(data.data.length);
      renderedItems.forEach((item, index) => {
        const {queryByText} = within(item);
        // 좋아요 다시 원래대로 돌아왔는지 테스트
        expect(
          queryByText(
            data.data[index].Like ? data.data[index].Like.toString() : '좋아요',
          ),
        ).toBeTruthy();
      });
    });
  });
});
