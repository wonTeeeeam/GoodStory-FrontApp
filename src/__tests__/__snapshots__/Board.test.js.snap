// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board 테스트 스냅샷 테스트 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#e9e9e9",
        "flex": 0.04,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={Object {}}
    >
      <TouchableOpacity
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginRight": 31.428571428571427,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#696969",
              }
            }
          >
            최신순
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 31.428571428571427,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <Modal
        animationType="none"
        deviceHeight={null}
        deviceWidth={null}
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        panResponderThreshold={4}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={false}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFAFA",
        "flex": 0.96,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
